File: src/main/java/io/github/slimefunguguproject/bump/utils/Patterns.java
package io.github.slimefunguguproject.bump.utils;

import java.util.regex.Pattern;

import lombok.experimental.UtilityClass;

/**
 * This class contains some {@link Pattern} used in Bump.
 *
 * @author ybw0014
 */
@UtilityClass
public final class Patterns {
    public static final Pattern MINECRAFT_NAMESPACEDKEY = Pattern.compile("minecraft:[a-z0-9/._-]+");
    public static final Pattern BUMP_TAG = Pattern.compile("#bump:[a-z_]+");
}

File: src/main/kotlin/io/github/slimefunguguproject/bump/utils/constant/Patterns.kt
package io.github.slimefunguguproject.bump.utils.constant

import java.util.regex.Pattern

object Patterns {
    val MINECRAFT_NAMESPACEDKEY: Pattern = Pattern.compile("minecraft:[a-z0-9/._-]+")
    val BUMP_TAG: Pattern = Pattern.compile("#bump:[a-z_]+")
    val BUMP_TAG_CONFIG: Pattern = Pattern.compile("bump:[a-z_]+")
}
